subxt/examples/unstable_light_client.rs

```diff
@@ -0,0 +1,150 @@
+//! This is a comprehensive example that utilizes the Light Client
+//! for a subset of subxt requests.
+//!
+//! Includes:
+//!  - Subscribes to all finalized blocks using the old RPC method
+//!  - Subscribes to the head of the chain using the new `chainHead` RPC method
+//!  - Dynamically query constants
+//!  - Dynamically decode the events of the latest block
+//!  - Various RPC calls to ensure proper shape of the response.
+//!
+//! # Note
+//!
+//! This feature is experimental and things might break without notice.
+
+use futures::StreamExt;
+use std::sync::Arc;
+use subxt::{
+    rpc::{types::FollowEvent, LightClientBuilder},
+    OnlineClient, PolkadotConfig,
+};
+
+// Generate an interface that we can use from the node's metadata.
+#[subxt::subxt(runtime_metadata_path = "../artifacts/polkadot_metadata_small.scale")]
+pub mod polkadot {}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::fmt::init();
+
+    // Create a light client from the provided chain spec.
+    // Note: this connects to the live polkadot chain.
+    let light_client = LightClientBuilder::new()
+        .chain_spec(include_str!("../artifacts/polkadot_spec.json"))
+        .build()
+        .await?;
+    let api = OnlineClient::<PolkadotConfig>::from_rpc_client(Arc::new(light_client)).await?;
```

Comment:
I'm wondering about how this is exposed.

Right now the lightclient stuff is an implementation of `RpcClientT`, but I wonder if it's useful to expose it at that level, or whether it'd be better if we exposed `LightClient<T>` right next to `OnlineClient<T>` and `OfflineClient<T>` in the `subxt::client` module (and when it's stable we could expose it right at the top level too). 

Ie we could have an itnerface like:

```rust
// subxt/client/light_client.rs:

pub struct LightClientBuilder {...}

pub struct LightClient<T>(OnlineClient<T>);

impl OnlineClient<T> for LightClient<T> {...}
impl OfflineClient<T> for LightClient<T> {...}

// usage:

use subxt::client::{ LightClient, LightClientBuilder };

let api: LightClient<PolkadotConfig> = LightClientBuilder::new()
    .chain_spec(include_str!("../artifacts/polkadot_spec.json"))
    .build::<PolkadotConfig>()
    .await?;
```

Then the fact that it's internally just implementing `RpcClientT` is just an implementation detail and doesn't really matter. Also lets us change what sort of things we expose on `LightClient<T>` in the future if we want etc.

Reply:
(and just to add; one nice thing about exposing a very minimum interface like this, rather than exposing any of the underlying RpcClientT impl etc, is that when the WASM version lands, we can just swap this `LightClient<T>` struct out with the wasm impl of it (if need by) and the rest of the interface hopefully remains identical :))